app/_hub/kong-inc/acme/index.md

```diff
@@ -0,0 +1,213 @@
+---
+name: ACME
+publisher: Kong Inc.
+version: 0.2.0
+
+source_url: https://github.com/Kong/kong-plugin-acme
+
+desc: Let's Encrypt and ACMEv2 integration with Kong
+description: |
+  This plugin allows Kong to apply cerificates from Let's Encrypt or any other ACMEv2 service and serve dynamically. Renew is handled with a configurable threshold time.
+
+type: plugin
+categories:
+  - security
+
+kong_version_compatibility:
+    community_edition:
+      compatible:
+        - 2.0.x
+    enterprise_edition:
+      compatible:
+
+params:
+  name: acme
+  api_id: false
+  service_id: false
+  route_id: false
+  consumer_id: false
+  protocols: ['http', 'https', 'tcp', 'tls', 'grpc', 'grpcs']
+  dbless_compatible: yes
+  config:
+    - name: account_email
+      required: yes
+      default:
+      description: |
+        The account identifier, can be reused in different plugin instance.
+    - name: api_uri
+      required: false
+      default: "`https://acme-v02.api.letsencrypt.org`"
+      description: |
+        The ACMEv2 API endpoint to use, user might use [Let's Encrypt staging environemnt](https://letsencrypt.org/docs/staging-environment/) during testing.
+    - name: cert_type
+      required: false
+      default: "`rsa`"
+      description: |
+        The certificate type to create, choice of `"rsa"` for RSA certificate or `"ecc"` for EC certificate.
+    - name: domains
+      required: false
+      default: "`[]`"
+      description: |
+        The list of domains to create certificate for. Wildcard domain like `*.example.com` is also supported. Regex pattern is not supported.
+    - name: renew_threshold_days
+      required: false
+      default: "`14`"
+      description: |
+        Days before expire to renew the certificate.
+    - name: storage
+      required: false
+      default: "`shm`"
+      description: |
+        The backend storage type to use, choice of `"kong"`, `"shm"`, `"redis"`, `"consul"` or `"vault"`. In dbless mode, `"kong"` storage is unavailable.
+    - name: storage_config
+      required: false
+      default:
+      description: |
+        Storage configs for each backend storage. See below for its default value.
+    - name: tos_accepted
+      required: false
```

Comment:
This should be always required?

Reply:
We should ask user to always supply this as `true`.

Reply:
We will check this value in schema if user is using let's encrypt and tos_accepted is not set to `true`. I feel it will be more user friendly when user omits this value to show `you must set this to true` instead of `setting this value is required`.